src/client/serialize/daos_serialize.c

```diff
@@ -101,48 +96,41 @@ serialize_str(hid_t file_id, struct daos_prop_entry *entry,
 	hid_t	usr_attr = 0;
 
 	if (entry == NULL || entry->dpe_str == NULL) {
-		rc = -DER_NOMEM;
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_INVAL);
 	}
 
 	attr_dims[0] = 1;
 	attr_dtype = H5Tcopy(H5T_C_S1);
 	if (attr_dtype < 0) {
-		rc = -DER_MISC;
 		D_ERROR("failed to create datatype\n");
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_MISC);
 	}
 	status = H5Tset_size(attr_dtype, strlen(entry->dpe_str) + 1);
 	if (status < 0) {
-		rc = -DER_MISC;
 		D_ERROR("failed to set datatype size\n");
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_MISC);
 	}
 	status = H5Tset_strpad(attr_dtype, H5T_STR_NULLTERM);
 	if (status < 0) {
-		rc = -DER_MISC;
 		D_ERROR("failed to set string pad on datatype\n");
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_MISC);
 	}
 	attr_dspace = H5Screate_simple(1, attr_dims, NULL);
 	if (attr_dspace < 0) {
-		rc = -DER_MISC;
 		D_ERROR("failed to create dataspace\n");
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_MISC);
 	}
 
 	usr_attr = H5Acreate2(file_id, prop_str, attr_dtype, attr_dspace,
 			      H5P_DEFAULT, H5P_DEFAULT);
 	if (usr_attr < 0) {
-		rc = -DER_MISC;
 		D_ERROR("failed to create attribute\n");
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_MISC);
 	}
 	status = H5Awrite(usr_attr, attr_dtype, entry->dpe_str);
 	if (status < 0) {
-		rc = -DER_IO;
 		D_ERROR("failed to write attribute "DF_RC"\n", DP_RC(rc));
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_MISC);
```

Comment:
Should this one be DER_MISC or DER_IO? It looks like it was DER_IO before, so just curious